shared/js/ui/base/localize.es6.js

```diff
@@ -1,37 +1,30 @@
 import i18next from 'i18next'
 import ICU from 'i18next-icu'
-import config from '../../../data/constants.js'
 
-const supportedLocales = config.supportedLocales
+const localeResources = require('../../../locales/*/*.json', { mode: 'list' })
 
-function buildLocalesList () {
-    const resources = {}
-    // hard coding for now till we figure out dynamic bundling
-    resources.en = {
-        shared: require('../../../locales/en/shared.json'),
-        options: require('../../../locales/en/options.json')
-    }
-    return resources
-}
-
-function getLocale () {
-    // truncate locale by dropping country suffix
-    let browserLocale = chrome.i18n ? chrome.i18n.getUILanguage() : navigator.language
-    browserLocale = browserLocale.split('-')[0]
+const resources = localeResources.reduce((mapping, { name, module }) => {
+    const [locale, namespace] = name.split('/')
+    mapping[locale] = mapping[locale] || {}
+    mapping[locale][namespace] = module
+    return mapping
+}, {})
```

Comment:
https://www.i18next.com/how-to/add-or-load-translations#lazy-load-in-memory-translations this seems a simpler option and removes the need for dynamic requires and loading all the files.